tests/ga/test_extension_telemetry.py

```diff
@@ -0,0 +1,539 @@
+                events_dir = os.path.join(conf.get_ext_log_dir(), ext_name, EVENTS_DIRECTORY)
+                fileutil.mkdir(events_dir)
+                shutil.copy(test_events_file_path, events_dir)
+
+            extension_names[ext_name] = no_of_good_events
+
+        return extension_names
+
+    @staticmethod
+    def _get_no_of_events_from_body(body):
+        return body.count("</Event>")
+
+    @staticmethod
+    def _replace_in_file(file_path, replace_from, replace_to):
+
+        with open(file_path, 'r') as f:
+            content = f.read()
+
+        content = content.replace(replace_from, replace_to)
+
+        with open(file_path, 'w') as f:
+            f.write(content)
+
+    @staticmethod
+    def _get_handlers_with_version_from_event_body(event_body):
+        event_with_name_and_versions = defaultdict(list)
+        regex_pattern = r'<Param Name="EventName" Value="(.+?)-(.+?)" .* \/>'
+        for body in event_body:
+            body = TestExtensionTelemetryHandler._get_ustr_from_event_body(body)
+            xml_doc = textutil.parse_doc(body)
+            events = textutil.findall(xml_doc, "Event")
+
+            for event in events:
+                # Since this is a telemetry event, we should always have an EventName param in the event
+                handler_name, version = re.search(regex_pattern, event.childNodes[0].nodeValue).groups()
+                event_with_name_and_versions[handler_name].append(version)
+
+        return event_with_name_and_versions
+
+    @staticmethod
+    def _get_param_value_from_event_body_if_exists(event_body, param_name):
+        regex_pattern = r'<Param Name="{0}" Value=(\"|\')(?P<value>.+?)(\"|\') .* \/>'.format(param_name)
+        param_values = []
+
+        for body in event_body:
+            body = TestExtensionTelemetryHandler._get_ustr_from_event_body(body)
+            xml_doc = textutil.parse_doc(body)
+            events = textutil.findall(xml_doc, "Event")
+
+            for event in events:
+
+                regex = re.search(regex_pattern, event.childNodes[0].nodeValue)
+                if regex is None:
+                    continue
+                param_values.append(regex.group('value'))
+
+        return param_values
+
+
+    @contextlib.contextmanager
+    def _create_extension_telemetry_processor(self):
+
+        def http_post_handler(url, body, **__):
+            if self.is_telemetry_request(url):
+                extension_telemetry_processor.event_body.append(body)
+                return MockHttpResponse(status=200)
+            return None
+
+        with mock_wire_protocol(DATA_FILE, http_post_handler=http_post_handler) as protocol:
+            protocol_util = MagicMock()
+            protocol_util.get_protocol = Mock(return_value=protocol)
+            extension_telemetry_processor = ProcessExtensionTelemetry(protocol_util) # get_extension_telemetry_processor(protocol_util)
```

Comment:
Is this a leftover comment?

Reply:
Ohh yes, thanks for catching it! Removed!